teams_rules.go

```diff
@@ -57,6 +57,12 @@ type TeamsRuleSettings struct {
 
 	// Action taken when an untrusted origin certificate error occurs in a http allow rule
 	UntrustedCertSettings *UntrustedCertSettings `json:"untrusted_cert"`
+
+	// Specifies that a resolver policy should use Cloudflare's DNS Resolver.
+	ResolveDnsThroughCF bool `json:"resolve_dns_through_cloudflare,omitempty"`
```

Comment:
as the JSON is `resolve_dns_through_cloudflare`, let's keep the struct matching it.

also, per https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#booleans (and eventually semgrep checks when the queue clears), we'll want to use `*bool`, not `bool` here.